### ISSUE TITLE
Range Sliders in the Inspector do not Assign Input Focus to their Associated Edit Control

### ISSUE THREAD

**Issue by jtwine** [godotengine/godot#80584]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.22621 - Vulkan (Compatibility) - NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.14.7311) - Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 Threads)

### Issue description

(Similar issues with the sliders are mentioned [here](https://github.com/godotengine/godot/issues/76620).
If you are in the Inspector and are tweaking values for something, as you navigate between controls and change values, if you decide to use a control's associated _slider control_ to adjust a value, the intended value is changed correctly (the correct edit control is updated), but if you are having trouble getting the exact value you want and decide to just type it in, whatever you type will be treated as input to the previously-focused input-capable control.

Similar to the behavior of _spin button_ controls in Windows, when you activate a control's "buddy control," (e.g. spin buttons, or the range slider in Godot) focus should shift to associated parent control.  This is what the user would normally expect as they are effectively _clicking in the client area of a control of interest_. This behavior can also be confirmed with with comboboxes - when you click the drop-down arrow, focus changes to the combobox.

### Steps to reproduce

1. With any project open, add and/or select a **Node2D** node.
2. Navigate to the Inspector and drop open the **Transform** pane.
3. Click into the _Position x_ control and enter a value.  
4. Click on the slider for _Rotation_ and use it to change the rotation.  Now type **45** as if you were trying to change the rotation to 45 degrees.  

Instead of **Rotation** changing, the value for **_Position x_** will be changed instead.  This is not the expected behavior.  (Also, if the previous control was a button, <**SPACE**> or <**ENTER**> would activate the button as well.)

This problem can be replicated with other slider controls as well, for different kinds of Nodes/objects.

### Minimal reproduction project

N/A

**Comment by Alex2782** [comment]
I could not reproduce the problem. Could you also make a video of exactly how you do it?

`Godot v4.1.1.stable - macOS 13.5.1 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)`

https://github.com/godotengine/godot/assets/41921395/26f334ba-8c8b-4e14-a2e7-6975f54f0028

**Comment by BlueCube3310** [comment]
> I could not reproduce the problem. Could you also make a video of exactly how you do it?

I think OP meant that using the SpinBox's slider does not automatically focus the SpinBox itself (which can be seen in the video). The user still has to click it to type the value manually, even after using the slider.

**Comment by jtwine** [comment]
> > I could not reproduce the problem. Could you also make a video of exactly how you do it?
> 
> I think OP meant that using the SpinBox's slider does not automatically focus the SpinBox itself (which can be seen in the video). The user still has to click it to type the value manually, even after using the slider.

Correct.  If you look at the latter part of the video, I focus a control above the spin/slider control and then I click on and drag the slider, but focus _remains on the earlier control_ - it does not move to the control "attached" to the slider.

The slider and its "buddy" control (to use Windows parlance) should be treated as a single control: focusing one allows _both_ to accept input, so you should be able to type a value in after dropping the slider if you cannot get the exact value your want.  Likewise, if you have clicked in the associated edit control, you should be able to move the slider in steps (preconfigured increments)using the arrow keys.
